server/controllers/report.js

```diff
@@ -71,10 +105,38 @@ function list(req, res, next) {
  * GET /reports/archive/:uuid
  */
 function sendArchived(req, res, next) {
-  // @TODO This will have to factor in the type of report - report uuid can be looked up in `saved_report` table
-  let reportPath = path.resolve(path.join(__dirname, '../reports/'));
-  let extension = '.pdf';
-  let reportUuid = req.params.uuid;
+  lookupArchivedReport(req.params.uuid)
+    .then(report => {
+      res.sendFile(report.link);
+    })
+    .catch(next)
+    .done();
+}
 
-  res.sendFile(reportPath.concat('/', reportUuid, extension));
+
+/**
+ * @function deleteArchived
+ *
+ * @description
+ * Deletes a report from the server.  This cleans up both the record of the
+ * report stored in saved_report and the file stored on the disk.
+ *
+ * DELETE /reports/archive/:uuid
+ */
+function deleteArchived(req, res, next) {
+  let filePath;
+
+  lookupArchivedReport(req.params.uuid)
+    .then(report => {
+      filePath = report.link;
+      return db.exec('DELETE FROM saved_report WHERE uuid = ?;', [ db.bid(req.params.uuid) ]);
+    })
+    .then(() => {
+      fs.unlink(filePath, err => {
+        if (err) { return next(err); }
```

Comment:
My logic was that if someone deletes a report, it means they no longer want to see it in their view.  Deleting the record will remove it from the grid, which accomplishes this.

Space might be an issue if `fs.unlink` misbehaves consistently.  But space is cheap in general, and we can always run some sort of `cron` job if this becomes a huge issue.

The alternative is worth considering: what happens if the file not being deleted blocks the record being deleted?  Potentially because of some weird file system bug that corrupted the file or a permissions issue?  I think I would prefer allowing the user to remove unused/unnecessary reports from their view (even if they persist on disk) rather than block the deletion because the file system is failing.

What do you think?